src/query/executor.rs

```diff
@@ -141,8 +141,10 @@ pub fn start_query(
                         Box::new(execute_test_multiply(ctx, input).await) as Box<dyn Result>
                     }
                     QueryType::IPA(config) => {
-                        let input = input.align(IPAInputRow::<Fp31>::SIZE_IN_BYTES);
-                        Box::new(execute_ipa(ctx, config, input).await) as Box<dyn Result>
+                        let input = input.align(IPAInputRow::<Fp31, BitArray40>::SIZE_IN_BYTES);
+                        // TODO: how to know `BitArray` type?
```

Comment:
do we need to add another value to `IPAQueryConfig` here?

Reply:
if so, get ready for exponential match statements...

Reply:
looks like we will try to make a `MatchKey` alias to use in place of `BitArray40`